### DISCUSSION TITLE
brew postinstall python@3.9 fails: Illegal instruction (core dumped)

### DISCUSSION THREAD

**Discussion by mlavi** [Homebrew/discussions#832] [Linux] [answered]
Hello, I have an older AMD  Phenom II X4 955 which might be a source of trouble, but Linuxbrew Python 3 hasn't been an issue on this system before!

brew config; brew doctor; brew postinstall --debug python@3.9;  brew reinstall python@3.9 [logs are in my Gist](https://gist.github.com/mlavi/ce330b2631de41575b1573e943539c8d)

Of course, all my Python3 dependent Linuxbrew package are now core-dumping. I've added tail of /var/log/{syslog,kern.log} which show the same thing:
Feb 16 20:08:35 marimayhem kernel: [ 5438.861829] traps: python3[26630] trap invalid opcode ip:7f2ceadb3db5 sp:7ffc5ad515f0 error:0 in libpython3.9.so.1.0[7f2ceac41000+373000]

Thanks for any help or insight to troubleshoot further!

**Comment by jonchang** [comment, marked as answer]
We don't support your processor for bottles on Linux as our minimum microarchitecture corresponds to Intel "core2" (https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html). You'll need to build everything from source with `brew install -s python@3.9`. Likely some change in the Python build process is now taking advantage of newer instructions and your Phenom doesn't recognize it. "Bulldozer" and upwards should work fine.

**Reply by mlavi** [reply]
Thanks for the prompt reply! Upgrading HW research begins. :)

**Reply by danielnachun** [reply]
This revealed something quite surprising to me: while the first Core2 processors were released in 2006, the first AMD processors to support SSSE3, the newest instruction set in Core2, were not released until 2011!  It looks like the oldest AMD CPUS with SSSE3 support are the "Bulldozer" and "Bobcat" architectures: https://en.wikipedia.org/wiki/SSSE3.  I'll take a look to see if there is a place we should add this to documentation.

**Reply by fmassart** [reply]
Thank you for this answer, I was meeting the same problem with the same processor.
Cheers
